recipe/meta.yaml

```
@@ -10,60 +10,71 @@ source:
   sha256: d3e8aa2e1f4092dfe8158523a4c9508f0759a400e8cca8a711669361ce23e883
 
 build:
+  number: 0
+  # No dependencies for py39
+  skip: true  # [py<310]
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
   entry_points:
     - glue = glue_qt.main:main
-  script: {{ PYTHON }} -m pip install . -vv
-  string: win_pyh{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [win]
-  string: unix_pyh{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [unix]
-  number: 1
-  osx_is_app: true  # [osx]
 
 requirements:
   host:
-    - python
-    - setuptools-scm
     - pip
+    - python
+    - setuptools
+    - setuptools_scm
   run:
-
-    # The following is needed to make sure that the package works as a GUI
-    # application (glue needs to be run with python.app, not python)
-    - python.app  # [osx]
-
     - python
     - glue-core >=1.15.0
     - numpy >=1.17
     - matplotlib-base >=3.2
     - scipy >=1.1
     - echo >=0.6
-    - astropy-base >=4.0
+    - astropy >=4.0
     - setuptools >=30.3.0
     - qtpy >=1.9
     - ipython >=4.0
     - ipykernel >=4.0,!=5.0.0,!=5.1.0
     - qtconsole >=4.3,!=5.4.2
     - pvextractor >=0.2
-    - importlib-resources >=1.3
-    - importlib-metadata >=3.6
-    - pyqt
+    - importlib-resources >=1.3  # [py<39]
+    - importlib-metadata >=3.6  # [py<310]
+  run_constrained:
+    - pyqt >=5.14
+
+{% set tests_to_skip = "_not_a_real_test" %}
+# AttributeError: 'Axes' object has no attribute 'coords'
+{% set tests_to_skip = tests_to_skip + " or test_load_viewers_04" %}  # [py<311]
+{% set tests_to_skip = tests_to_skip + " or test_load_pixel_components_07" %}  # [py<311]
+
 test:
   imports:
     - glue_qt
     - glue_qt.app
+  requires:
+    - pip
+    - pytest
+    - pyqt >=5.14
```

```suggestion
    - pyqt
```
Since the package itself has a constraint on it you should be able to just let conda resolve the version it wants to use.